assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useMedia3pApi } from './api';
+
+export default function useFetchMediaEffect({
+  provider,
+  selectedProvider,
+  pageToken,
+  fetchMediaStart,
+  fetchMediaSuccess,
+  fetchMediaError,
+}) {
+  const {
+    actions: { listMedia },
+  } = useMedia3pApi();
+
+  useEffect(() => {
+    async function fetch() {
+      fetchMediaStart({ provider, pageToken });
+      try {
+        const { media, nextPageToken } = await listMedia({
+          provider,
+          pageToken,
+        });
+        fetchMediaSuccess({ provider, media, nextPageToken });
```

Comment:
Should we be using `function` rather than arrow functions here? 

Maybe it'll be simpler to just to put the if statement to an if statement at the top rather than extract fetch as a function. 

i.e. if (provider !== selectedProvider)  return; 
/// put the content of "fetch" here ///

Reply:
The "useEffect(() => {" lambda isn't allowed to be an async function.

https://medium.com/javascript-in-plain-english/how-to-use-async-function-in-react-hook-useeffect-typescript-js-6204a788a435